src/Apps/Order/Routes/Shipping2/ShippingContext.ts

```diff
@@ -1,20 +1,342 @@
-import { FulfillmentValues } from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
-import { createContext, useContext } from "react"
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from "react"
+import { FormikProps } from "formik"
+import { ShippingProps, ShippingRouteStep } from "Apps/Order/Routes/Shipping2"
+import {
+  FulfillmentValues,
+  ShipValues,
+} from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
+import {
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+  FulfillmentType,
+  ShippingAddressFormValues,
+} from "Apps/Order/Routes/Shipping2/shippingUtils"
+import { SavedAddressType } from "Apps/Order/Utils/shippingUtils"
+import { extractNodes } from "Utils/extractNodes"
+import { ALL_COUNTRY_CODES, EU_COUNTRY_CODES } from "Components/CountrySelect"
+import { COUNTRIES_IN_EUROPEAN_UNION } from "@artsy/commerce_helpers"
+// TODO: Duplicated list ^
 
+type FulfillmentHelpers = Pick<
+  FormikProps<FulfillmentValues>,
+  "handleSubmit" | "isValid" | "values"
+>
 export interface ShippingContextProps {
-  lockShippingCountryTo: "EU" | string | null
-  shipsFrom: string
-  availableShippingCountries: string[]
-  requiresArtsyShippingTo: (shipTo: string) => boolean
-  selectedSavedAddressId: string | null
-  fulfillmentDetails: FulfillmentValues["attributes"] | null
-  fulfillmentType: FulfillmentValues["fulfillmentType"] | null
-  isArtsyShipping?: boolean
-  shippingQuotes?: Array<{ id: string; isSelected: boolean }>
+  savedOrderData: {
+    lockShippingCountryTo: "EU" | string | null
+    shipsFrom: string
+    availableShippingCountries: string[]
+    requiresArtsyShippingTo: (shipTo: string) => boolean
+    selectedSavedAddressId: string | null
+    fulfillmentDetails: FulfillmentValues["attributes"] | null
+    fulfillmentType: FulfillmentValues["fulfillmentType"] | null
+    isArtsyShipping?: boolean
+    shippingQuotes?: Array<{ id: string; isSelected: boolean }>
+  }
+  initialValues: {
+    fulfillmentDetails: FulfillmentValues
```

Comment:
is it `fulfilmentDetails` or `FulfilmentValues`? 

Again, 1:2 complexity. 

Every new variable in this PR needs to be audited for 1:1 correctness 🙏 The only way we'll reduce complexity in the shipping route is by literally reducing complexity.

Reply:
`fulfillmentDetails` refers to what the form/component is named, while `FulfillmentValues` is the interface of the form values

Reply:
if `FulfilmentValues` is the type for `fulfilmentDetails`, how does that make any sense? One doesn't typically write `foo: Bar`; one writes `foo: Foo` and `bar: Bar`, and this is particularly true in highly sensitive areas that deal with money.

Reply:
Seeing the follow-up component below, this makes more sense 👍